contracts/SimpleSwap.sol

```diff
@@ -62,50 +61,63 @@ contract SimpleSwap {
   /// @param beneficiary the beneficiary of the cheque
   /// @param serial the serial number of the cheque
   /// @param amount the (cumulative) amount of the cheque
+  /// @param timeout the check can be cashed timeout seconds in the future
   function _submitChequeInternal(address beneficiary, uint serial, uint amount, uint timeout) internal {
+    ChequeInfo storage cheque = cheques[beneficiary];
     /* ensure serial is increasing */
-    ChequeInfo storage info = cheques[beneficiary];
-    require(serial > info.serial);
-
+    require(serial > cheque.serial,
+    "SimpleSwap: invalid serial");
     /* update the stored info */
-    info.serial = serial;
-    info.amount = amount;
-    /* the check can be cashed timeout seconds in the future */
-    info.timeout = now + timeout;
-
+    cheque.serial = serial;
+    cheque.amount = amount;
+    cheque.timeout = now + timeout;
     /* the channel participants should watch to this event to find out if an older cheque is being submitted */
     emit ChequeSubmitted(beneficiary, serial, amount);
   }
 
-  /// @notice submit a cheque
+  /// @notice submit a cheque by the owner
   /// @param beneficiary the beneficiary of the cheque
   /// @param serial the serial number of the cheque
   /// @param amount the (cumulative) amount of the cheque
-  /// @param sig signature of the owner
-  function submitCheque(address beneficiary, uint serial, uint amount, uint timeout, bytes memory sig) public {
-    /* only allow beneficiary to submit this, otherwise the owner could block cash out by regulary sending 1 wei cheques and resetting the timeout */
-    /* unfortunately this breaks watchtowers, so the timeout mechanism should be changed */
-    require(msg.sender == beneficiary);
-    /* verify signature of the owner */
-    require(owner ==  recover(chequeHash(address(this), beneficiary, serial, amount, timeout), sig));
-    /*  amount needs to be larger. since this can only be called by the beneficiary this is probably not necessary */
-    require(amount > cheques[beneficiary].amount);
```

Comment:
why is this check gone? why do we not need it anymore?

Reply:
We still need this, but it is implemented in a slightly different way:
The rationale behind is, is that in order to submit a cheque, this cheque must be seen by both the owner and the beneficiary. We implement this in three different ways:
- In case the owner sends it (submitChequeOwner): no explicit ownersig is needed, as he already sends it.
- In case the beneficiary sends it (submitChequeBeneficiary): no explicit beneficiarySig is needed
- In case any other party wants to send it (submitCheque), we need an explicit signature from both the owner and the beneficiary.

Submitcheque on it's own would be sufficient, but in order to save gas, the owner and beneficiary can send the cheque to their 'own' functions (see test case on gas costs in SimpleSwap.js). 

Generally, the code is easier to reason about this way and will cost less gas than the previous version.

In the submitChequeBeneficiary we could remove the explicit require statement on ``msg.sender == beneficiary``, as the beneficiary is part of the field where the owner has given his signature on. We could remove the require statement on ``amount > cheque[beneficiary].amount`` as in all of the three functions we require that both the beneficiary and the owner give their consent, hence: who are we to dictate which price they want to set?